cpp/src/text/subword/load_hash_file.cu

```diff
@@ -40,11 +41,11 @@ namespace {
 struct get_codepoint_metadata_init {
   rmm::cuda_stream_view stream;
 
-  codepoint_metadata_type* operator()() const
+  rmm::device_uvector<codepoint_metadata_type>* operator()() const
   {
-    codepoint_metadata_type* table =
-      static_cast<codepoint_metadata_type*>(rmm::mr::get_current_device_resource()->allocate(
-        codepoint_metadata_size * sizeof(codepoint_metadata_type), stream));
+    auto table_vector =
+      new rmm::device_uvector<codepoint_metadata_type>(codepoint_metadata_size, stream);
```

Comment:
I'm not sure I've grokked how/where the `device_uvector` is `delete`d.

Reply:
It is not deleted. The table is too big for `__constant__` memory so this code builds it manually into device memory instead.

Reply:
For my education, how much device memory might this allocate when called? The concern I had was regarding the effect this permanent allocation would have on the Spark side:

1. JVM reuse: It's unlikely that the JVM might be reused with a non-`libcudf` task in Spark. So we don't have to worry about this being reallocated in a separate task.
2. Since this initialization happens from a function static, the allocation only happens when `subword_tokenize` is exercised (i.e. possibly well after startup). Would fragmentation be a concern? Maybe that depends on how big the device allocation is.

In any case, the permanent device allocation is already happening. This PR only changes how the allocation happens (`get_current_device_resource()->allocate()` vs `device_uvector<>::ctor`). This concern shouldn't hold up the current PR.

Reply:
The total size of the tables is about (1114112 * 4) + (119233 * 8) = ~5.4MB